app/controllers/products_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ProductsController < ApplicationController
+  def new
+    @product = Product.new
+    @vendor = Vendor.find(params[:vendor_id])
+  end
+
+  def create
+    product = Product.create(product_params) do |p|
+      p.vendor = Vendor.find(params[:vendor_id])
+    end
+    redirect_to product.vendor
+  end
+
+  def edit
+    @product = Product.find(params[:id])
```

Comment:
As with `MarketsController`, because we set `@product` and `@vendor` in many of the controller methods, it would be somewhat more DRY if we moved that code into a `before_action` (maybe using if statements to determine if we need to set `@vendor`).